toffy/panel_utils.py

```diff
@@ -0,0 +1,140 @@
+               'chan_147', 'chan_148', 'chan_149', 'chan_150', 'chan_151', 'chan_152', 'chan_153',
+               'chan_154', 'chan_155', 'chan_156', 'chan_157', 'chan_158', 'chan_159', 'chan_160',
+               'chan_161', 'chan_162', 'chan_163', 'chan_164', 'chan_165', 'chan_166', 'chan_167',
+               'chan_168', 'chan_169', 'chan_170', 'chan_171', 'chan_172', 'chan_173', 'chan_174',
+               'chan_175', 'chan_176', 'Au']})
+
+
+def drop_duplicate_masses(panel):
+    """ Check a panel df for duplicate mass values and return a unique mass panel with the
+        target names combined
+    Args:
+        panel (pd.DataFrame): panel dataframe with columns Mass and Target
+    Returns:
+        pd.DataFrame with no duplicate masses
+    """
+
+    # find the mass and target values of duplicate
+    duplicates = panel[panel['Mass'].duplicated(keep=False)]
+
+    # combine target names
+    for mass in duplicates['Mass'].unique():
+        duplicate_targets = panel.loc[panel['Mass'] == mass, 'Target']
+        target_name = ''.join(name + '_' for name in duplicate_targets)[:-1]
+        panel.loc[panel['Mass'] == mass, 'Target'] = target_name
+
+    # only keep one row for each mass value
+    unique_panel = panel.drop_duplicates(subset=['Mass'], keep='first')
+
+    return unique_panel
+
+
+def convert_panel(panel_path):
+    """
+    Converts the panel retrieved from ionpath into a necessary toffy format,
+    also adds necessary channels for analysis
+    Args:
+        panel_path (str): direct path to panel file
+    Returns:
+        panel (pd.DataFrame): detailing the Mass, Target, Start, and Stop values
+        also saves the above to file with name formatted "panel_name-toffy.csv"
+    """
+
+    # retrieve the panel name and directory it is contained in
+    panel_name = os.path.basename(os.path.splitext(panel_path)[0])
+    panel_dir = os.path.dirname(panel_path)
+
+    # read in metadata contained in the first few lines
+    col_index = 0
+    with open(panel_path) as r:
+        lines = r.readlines()
+        for index, line in enumerate(lines):
+            # line containing the column names we want
+            if len(line.split(',')) == 10:
+                col_index = index
+                break
+        r.close()
+
+    # load in the panel info to df
+    with open(panel_path, encoding='utf-8-sig') as r:
+        headers = [next(r) for i in list(range(col_index+1))]
+        cols = headers[col_index].split(',')
+        panel = pd.read_csv(r, sep=',', names=cols, index_col=False)
+        r.close()
+
+    # check for already correctly formatted panel, return as is
+    if list(panel.columns) == ['Mass', 'Target', 'Start', 'Stop'] or list(panel.columns) ==\
+            ['Mass', 'Target', 'Start', 'Stop\n']:
+        print(f'{panel_name}.csv has the correct toffy format. Loading in panel data.')
+        return panel
+
+    # retrieve column names, and original mass / target values
```

Comment:
Should add in a check if it's neither ionpath nor toffy panel format and fail with an error message